Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -1022,7 +1023,7 @@ Target Serializer::DecodeQueryTarget(
 
   return Query(std::move(path), std::move(collection_group),
                std::move(filter_by), std::move(order_by), limit,
-               std::move(start_at), std::move(end_at))
+               LimitType::None, std::move(start_at), std::move(end_at))
```

Comment:
`LimitType::None` doesn't seem right here if `limit` ends up with a value other than `Target::kNoLimit`.

Should this be `LimitType::First`?